📕 🏴‍☠️ Update docs and deprecations (#979)
SmartDataAnalytics/PyKEEN: docs/source/extending/models.rst

        def forward(self, h, r, t):
            return (h * r * t).sum(dim=-1)

    class DistMult(ERModel):
        def __init__(
            self,
            # When defining your class, any hyper-parameters that can be configured should be
            # made as arguments to the __init__() function. When running the pipeline(), these
            # are passed via the ``model_kwargs``.
            embedding_dim: int = 50,
            # All remaining arguments are simply passed through to the parent constructor. If you
            # want access to them, you can name them explicitly. See the pykeen.models.ERModel
            # documentation for a full list
            **kwargs,
        ) -> None:
            # since this is a python class, you can feel free to get creative here. One example of
            # pre-processing is to derive the shape for the relation representation based on the
            # embedding dimension.
            super().__init__(
                # Pass an instance of your interaction function. This is also a place where you can
                # pass hyper-parameters, such as the L_p norm, from the KGEM to the interaction function

>                 interaction=DistMultInteraction(),

Author: ```suggestion
                interaction=DistMultInteraction,
                # interaction_kwargs=dict(...),
```

                # Define the entity representations using a dict. By default, each

>                 # embedding is linear. You can use the ``shape`` kwarg to specify higher dimensional

Author: ```suggestion
                # embedding is a vector. You can use the ``shape`` kwarg to specify higher dimensional
```

Obsolete Tutorial: Extending the Interaction Models (Old-Style)
===============================================================
.. warning::

    This tutorial is no longer valid since we removed the ``EntityRelationEmbeddingModel`` class.

>     in PyKEEN 1.8.2. Use the ``ERModel`` class for custom models now.

Author: should we remove that part then?

Reviewer: Some of the ideas are good and we should port over to new ERmodels but rewriting docs is no fun and I had no motivation this morning (or now)

Reviewer: Done in 764c0a56

    The HPO rules are subject to change as they are tightly coupled to :mod:`optuna`,
    which since version 2.0.0 has introduced several new possibilities.

Implementing a model by instantiating :class:`pykeen.models.ERModel`
--------------------------------------------------------------------
Instead of creating a new class, you can also directly use the :class:`pykeen.models.ERModel`, e.g.

.. code-block:: python

    from pykeen.models import ERModel
    from pykeen.losses import BCEWithLogitsLoss

    model = ERModel(
        triples_factory=...,

>         loss=BCEWithLogitsLoss(),

Author: ```suggestion
        loss="BCEWithLogits",
```
